bugsnag-plugin-android-ndk/src/main/jni/utils/serializer/event_reader.c

```diff
@@ -402,22 +487,39 @@ void migrate_device_v1(bugsnag_report_v2 *report_v2, bugsnag_report_v3 *event) {
   bsg_strcpy(event->device.os_version, report_v2->device.os_version);
 
   // migrate legacy fields to metadata
-  bugsnag_event_add_metadata_bool(event, "device", "emulator",
-                                  report_v2->device.emulator);
-  bugsnag_event_add_metadata_double(event, "device", "dpi",
-                                    report_v2->device.dpi);
-  bugsnag_event_add_metadata_double(event, "device", "screenDensity",
-                                    report_v2->device.screen_density);
-  bugsnag_event_add_metadata_double(event, "device", "batteryLevel",
-                                    report_v2->device.battery_level);
-  bugsnag_event_add_metadata_string(event, "device", "locationStatus",
-                                    report_v2->device.location_status);
-  bugsnag_event_add_metadata_string(event, "device", "brand",
-                                    report_v2->device.brand);
-  bugsnag_event_add_metadata_string(event, "device", "networkAccess",
-                                    report_v2->device.network_access);
-  bugsnag_event_add_metadata_string(event, "device", "screenResolution",
-                                    report_v2->device.screen_resolution);
+  add_metadata_bool(&event->metadata, "device", "emulator",
+                    report_v2->device.emulator);
+  add_metadata_double(&event->metadata, "device", "dpi", report_v2->device.dpi);
+  add_metadata_double(&event->metadata, "device", "screenDensity",
+                      report_v2->device.screen_density);
+  add_metadata_double(&event->metadata, "device", "batteryLevel",
+                      report_v2->device.battery_level);
+  add_metadata_string(&event->metadata, "device", "locationStatus",
+                      report_v2->device.location_status);
+  add_metadata_string(&event->metadata, "device", "brand",
+                      report_v2->device.brand);
+  add_metadata_string(&event->metadata, "device", "networkAccess",
+                      report_v2->device.network_access);
+  add_metadata_string(&event->metadata, "device", "screenResolution",
+                      report_v2->device.screen_resolution);
+}
+
+void migrate_device_v2(bsg_device_info *output, bsg_device_info_v2 *input) {
+  output->api_level = input->api_level;
+  output->cpu_abi_count = input->cpu_abi_count;
+  memcpy(&output->cpu_abi, input->cpu_abi, sizeof(input->cpu_abi));
+  memcpy(&output->orientation, input->orientation, sizeof(input->orientation));
+  output->time = input->time;
+  memcpy(&output->id, input->id, sizeof(input->id));
```

Comment:
Would it be preferable to use `bsg_strncpy` here? I can't see anything obvious going wrong with using `memset` but it'd be more in keeping with how the rest of the migrations work IIRC

Reply:
Its functionally equivalent -  the benefit of using `memcpy` was not needing to switch functions when copying array types other than `char` (like `cpu_abi`).